src/SDK/Metrics/MeterProvider.php

```diff
@@ -0,0 +1,181 @@
+use OpenTelemetry\SDK\Resource\ResourceInfo;
+
+final class MeterProvider implements MeterProviderInterface
+{
+    private ?ContextStorageInterface $contextStorage;
+    private MetricFactoryInterface $metricFactory;
+    private ResourceInfo $resource;
+    private AttributesFactoryInterface $attributesFactory;
+    private ClockInterface $clock;
+    private InstrumentationScopeFactoryInterface $instrumentationScopeFactory;
+    /** @var MetricSourceRegistryInterface&MetricReaderInterface */
+    private MetricReaderInterface $metricReader;
+    private StalenessHandlerFactoryInterface $stalenessHandlerFactory;
+    private CriteriaViewRegistry $criteriaViewRegistry;
+    private ViewRegistryInterface $viewRegistry;
+    private MeterInstruments $instruments;
+
+    private bool $closed = false;
+
+    /**
+     * @param MetricSourceRegistryInterface&MetricReaderInterface $metricReader
+     */
+    public function __construct(
+        ?ContextStorageInterface $contextStorage,
+        ResourceInfo $resource,
+        ClockInterface $clock,
+        InstrumentationScopeFactoryInterface $instrumentationScopeFactory,
+        $metricReader,
+        AttributesFactoryInterface $attributesFactory,
+        StalenessHandlerFactoryInterface $stalenessHandlerFactory,
+        MetricFactoryInterface $metricFactory = null
+    ) {
+        $this->contextStorage = $contextStorage;
+        $this->metricFactory = $metricFactory ?? new StreamFactory();
+        $this->resource = $resource;
+        $this->attributesFactory = $attributesFactory;
+        $this->clock = $clock;
+        $this->instrumentationScopeFactory = $instrumentationScopeFactory;
+        $this->metricReader = $metricReader;
+        $this->stalenessHandlerFactory = $stalenessHandlerFactory;
+        $this->criteriaViewRegistry = new CriteriaViewRegistry();
+        $this->viewRegistry = new FallbackViewRegistry($this->criteriaViewRegistry, [
+            new ViewTemplate(
+                null,
+                null,
+                null,
+                self::defaultAggregation(),
+                self::defaultExemplarReservoir(),
+            ),
+        ]);
+        $this->instruments = new MeterInstruments();
+    }
+
+    public function getMeter(
+        string $name,
+        ?string $version = null,
+        ?string $schemaUrl = null,
+        iterable $attributes = []
+    ): MeterInterface {
+        if ($this->closed) {
+            return new NoopMeter();
+        }
+
+        return new Meter(
+            $this->contextStorage,
+            $this->metricFactory,
+            $this->resource,
+            $this->clock,
+            $this->attributesFactory,
+            $this->stalenessHandlerFactory,
+            $this->viewRegistry,
+            $this->metricReader,
+            $this->instruments,
+            $this->instrumentationScopeFactory->create($name, $version, $schemaUrl, $attributes),
+        );
+    }
+
+    public function registerView(
```

Comment:
Move to constructor? Currently views have to be configured in advance anyways / are not applied retroactively. [Meter Creation](https://github.com/open-telemetry/opentelemetry-specification/blob/793354957a11b839d677084d074a1ee11f474b73/specification/metrics/sdk.md#meter-creation):
> [...] SDK MUST provide a way to configure all options that are implemented by the SDK. This MAY be done at the time of MeterProvider creation if appropriate.